### PR TITLE
Add solution to qs 2 from sorting and Add qs 4 and solution from stacks

### PR DESCRIPTION
**Pull request by abdulaz1z**

### CODE REVIEW

**Comment by abdulaz1z** [comment]
@neoprez @Raymond-P check it out

File: src/sorting/SortingQuestions.java [hunk 1]
```diff
@@ -25,6 +25,34 @@
...
+        for (int i = 0; i < length; i++){
+            indexOfMinElement = i;  //assuming the ith element to be the smallest at start of the inner loop
+            for (int j = i + 1; j < length; j++){
+                if (arr[j] < arr[indexOfMinElement]){
+                    //updating with the next minimum value index
+                    indexOfMinElement = j;
+                }
+
+                //swapping the current minimum to be in the sorted portion of the array
+                temp = arr[i];
```
Comment (neoprez) [inline src/sorting/SortingQuestions.java]: In place swap must happen after the second loop. Insertion sort improves over bubble sort in the fact that the number of swaps is reduced. As of right now, you are swapping the elements after every comparison therefore it is still bubble sort.

Comment (nslehman) [reply src/sorting/SortingQuestions.java]: @neoprez thank you! I see the difference between bubble and selection

**Comment by abdulaz1z** [comment]
@neoprez Done! Thank you for the catch

**Comment by neoprez** [comment]
Looks good to me.

**Review by neoprez — APPROVED** [review]

**Pull request merged**